crates/win-api-wrappers/src/netmgmt.rs

```diff
@@ -0,0 +1,102 @@
+use std::ptr::null_mut;
+
+use anyhow::{bail, Result};
+use windows::Win32::Foundation::WIN32_ERROR;
+
+use crate::identity::account::Account;
+use crate::identity::sid::{RawSid, Sid};
+use crate::raw::Win32::NetworkManagement::NetManagement::{
+    NERR_Success, NetApiBufferFree, NetLocalGroupAddMembers, NetLocalGroupDelMembers, NetLocalGroupGetMembers,
+    LOCALGROUP_MEMBERS_INFO_0, MAX_PREFERRED_LENGTH,
+};
+use crate::raw::Win32::Security::WinBuiltinAdministratorsSid;
+use crate::utils::WideString;
+use crate::Error;
+
+pub fn add_local_group_member(group_name: &str, security_identifier: &Sid) -> Result<()> {
+    // SAFETY: It is safe to zero out the structure as it is a simple POD type.
+    let mut group_info = unsafe { core::mem::zeroed::<LOCALGROUP_MEMBERS_INFO_0>() };
+
+    let group_name = WideString::from(group_name);
+
+    let user_sid = RawSid::try_from(security_identifier)?;
+    group_info.lgrmi0_sid = user_sid.as_psid();
+
+    // SAFETY: All buffers are valid.
+    // WideString holds a null-terminated UTF-16 string, and as_pwstr() returns a valid pointer to it.
+    let rc =
+        unsafe { NetLocalGroupAddMembers(None, group_name.as_pcwstr(), 0, &group_info as *const _ as *const u8, 1) };
```

Comment:
Left this as-is for now, because I'm not sure I know the best way to follow your suggestion.

Is it the case that I need to break out an intermediate variable?

i.e.

`let group_name_ptr = &group_info as *const LOCALGROUP_MEMBERS_INFO_0;`

and then use `cast()` at the final call site, i.e.

`unsafe { NetLocalGroupDelMembers(None, group_name.as_pcwstr(), 0, group_name_ptr.cast(), 1) };`

But probably I misunderstand

Reply:
You can wrap with parenthesis:
```rust
(&group_info as *const _).cast::<u8>()
```

But splitting into several lines may be more readable in some cases, especially if we spell out the type like in your example.

Reply:
Thanks!